### PR TITLE
Integrate @vector-im/compound-web

### PR DESCRIPTION
**Pull request by germain-gg**
Fixes https://github.com/vector-im/element-web/issues/25905
For https://github.com/vector-im/element-web/issues/25883

To review alongside https://github.com/vector-im/element-web/pull/25906

## Checklist

-   [x] Tests written for new code (and old code if feasible)
-   [x] Linter and other CI checks pass
-   [x] Sign-off given on the changes (see [CONTRIBUTING.md](https://github.com/matrix-org/matrix-react-sdk/blob/develop/CONTRIBUTING.md))

---
This change is marked as an *internal change* (Task), so will not be included in the changelog.

### CODE REVIEW

File: test/components/views/rooms/__snapshots__/RoomHeader-test.tsx.snap [hunk 1]
```diff
@@ -10,9 +10,9 @@ exports[`Roomeader renders with no props 1`] = `
     >
       <div
         aria-level="1"
-        class="mx_RoomHeader_name"
         dir="auto"
         role="heading"
+        style="font: var(--cpd-font-body-lg-semibold); letter-spacing: var(--cpd-font-letter-spacing-body-lg);"
```
Comment (t3chguy) [inline test/components/views/rooms/__snapshots__/RoomHeader-test.tsx.snap:15]: This switching to inline styles is surprising, why are we bloating the DOM here?

File: src/components/views/rooms/RoomHeader.tsx [hunk 2]
```diff
@@ -40,10 +41,14 @@ export default function RoomHeader({ room, oobData }: { room?: Room; oobData?: I
         >
             {room && <DecoratedRoomAvatar room={room} oobData={oobData} avatarSize={40} displayBadge={false} />}
             <div className="mx_RoomHeader_info">
-                <div dir="auto" title={roomName} role="heading" aria-level={1} className="mx_RoomHeader_name">
+                <Body as="div" size="lg" weight="semibold" dir="auto" title={roomName} role="heading" aria-level={1}>
```
Comment (t3chguy) [inline src/components/views/rooms/RoomHeader.tsx:44]: Body is a really poor name IMO, could we import it as `BodyText` maybe?

Comment (germain-gg) [reply src/components/views/rooms/RoomHeader.tsx:44]: That probably won't be fixed in this pull request, but rather as part of github.com/vector-im/compound
But can definitely hear the argument that we want to change that name

File: test/components/views/rooms/__snapshots__/RoomHeader-test.tsx.snap (hunk 1 above)
Comment (germain-gg) [reply test/components/views/rooms/__snapshots__/RoomHeader-test.tsx.snap:15]: I agree with you, it's a bit odd. It's on my list of things to fix on Compound's side. There's more elegant ways to achieve the same thing. 
I however don't think we should block that pull request on this specific thing.

Comment (t3chguy) [reply test/components/views/rooms/__snapshots__/RoomHeader-test.tsx.snap:15]: > I however don't think we should block that pull request on this specific thing.

I agree in this instance, but I argue it is a blocker to switching to compound components for components used frequently in the DOM, given you get at most 1 room header this is fine

File: src/components/views/rooms/RoomHeader.tsx (hunk 2 above)
Comment (germain-gg) [reply src/components/views/rooms/RoomHeader.tsx:44]: https://github.com/vector-im/compound/issues/194

File: test/components/views/rooms/__snapshots__/RoomHeader-test.tsx.snap (hunk 1 above)
Comment (germain-gg) [reply test/components/views/rooms/__snapshots__/RoomHeader-test.tsx.snap:15]: https://github.com/vector-im/compound/issues/194

**Review by t3chguy — APPROVED** [review]
LGTM

**Pull request merged**